src/core/getProvider.ts

```diff
@@ -0,0 +1,34 @@
+import type { Provider } from 'ethers';
+import { FetchRequest, JsonRpcProvider, WebSocketProvider } from 'ethers';
+import shouldNeverHappen from '../utils/shouldNeverHappen';
+import appSettings from '../config/appSettings';
+
+let providerInstance: Provider | null = null;
+
+export default function getProvider(): Provider {
+  if (!providerInstance) {
+    const { rpcUrl, glifToken, pollingInterval } = appSettings;
+
+    if (rpcUrl.includes('node.glif.io')) {
+      const fetchRequest = new FetchRequest(rpcUrl);
+
+      fetchRequest.method = 'POST';
```

Comment:
this shouldn't be necessary, better remove it just in case it somehow might mess something up^^